azurerm/data_source_subscription_test.go

```diff
@@ -0,0 +1,79 @@
+package azurerm
+
+import (
+	"fmt"
+	"os"
+	"testing"
+
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/terraform"
+)
+
+func TestAccDataSourceAzureRMSubscription_current(t *testing.T) {
+	resourceName := "data.azurerm_subscription.current"
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:  func() { testAccPreCheck(t) },
+		Providers: testAccProviders,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccDataSourceAzureRMSubscription_currentConfig,
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttrSet(resourceName, "subscription_id"),
+					testCheckAzureRMSubscriptionId(resourceName),
+					resource.TestCheckResourceAttrSet(resourceName, "display_name"),
+				),
+			},
+		},
+	})
+}
+
+func TestAccDataSourceAzureRMSubscription_specific(t *testing.T) {
+	resourceName := "data.azurerm_subscription.specific"
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:  func() { testAccPreCheck(t) },
+		Providers: testAccProviders,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccDataSourceAzureRMSubscription_specificConfig(os.Getenv("ARM_SUBSCRIPTION_ID")),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttrSet(resourceName, "subscription_id"),
+					testCheckAzureRMSubscriptionId(resourceName),
```

Comment:
I think we can remove this in favour of checking via:

```
resource.TestCheckResourceAttr(resourceName, "subscription_id", "{Subscription ID Value}")
```

Reply:
The reason we can't do as suggested is that `{Subscription ID Value}` is obtained from the `ArmClient` for the running test step and this can't be done when the steps are declared (`testAccProvider.Meta().(*ArmClient)` is `nil` at this point).

Reply:
true, but we could instead check this against the `ARM_SUBSCRIPTION_ID` environment variable? It's not a blocker by any means and this otherwise LGTM :)